mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -453,11 +460,45 @@ public boolean removeClanAsEnemy(Transaction transaction, Clan clan) {
      * @return All clans this clan has currently marked as enemies
      */
     public Set<Clan> getEnemyClans(Transaction transaction) {
-        return transaction.getProperty(DSLContext.class)
+        var result = transaction.getProperty(DSLContext.class)
                 .select(CLANS_CLAN_ENEMIES.ENEMY_ID)
                 .from(CLANS_CLAN_ENEMIES)
                 .where(CLANS_CLAN_ENEMIES.CLAN_ID.eq(this.clanId))
-                .fetchSet((record) -> manager.getClan(transaction, record.get(CLANS_CLAN_ENEMIES.ENEMY_ID)));
+                .fetch();
+
+        Set<Clan> clans = new HashSet<>();
+
+        result.forEach(res -> {
+            Optional<Clan> op = manager.getClanById(transaction, res.get(CLANS_CLAN_ENEMIES.ENEMY_ID));
+
+            op.ifPresent(clans::add);
+        });
+
+        return clans;
+    }
+
+
+    /**
+     * Sets the owner of the clan to the user provided by the id, and
+     * if they are not a member of the clan they are added.
+     *
+     * The old owner remains a member of the clan.
+     *
+     * @param transaction tx
+     * @param id user id of the new clan owner
+     */
+    public void setOwner(Transaction transaction, int id) {
```

Comment:
```suggestion
    public void setLeader(Transaction transaction, SolarPlayer player) {
```
I have multiple suggestions:
1. Regarding naming, we should use "leader" rather than "owner" to be consistent with elsewhere.
2. For the API parameter, a SolarPlayer is more type-safe and API friendly than an integer ID.
3. A bug: the implementation needs to update `this.leader` (the cached leader) with the specified one
4. For maintainability, can you add an integration test for this method? The relevant infrastructure for integration tests with dataloader, and even in mod-clans, is already set up.
5. A bug: If the specified user is a member of the clan, the insert to the membership table will fail with a primary key violation.
6. Concerning unsafe behavior: Conversely, if the specified user is not a member of the clan, the insert to the membership table will silently add them to the clan. Can we instead check that the user is already a member of the clan?
7. For code cleanliness, can you avoid using transaction.getProperty(DSLContext.class) twice in the same method?

Reply:
sounds good

Reply:
Regarding number 5 - The documentation explicitly states that they will be added. Not unsafe?